pennylane_lightning/lightning_gpu/_state_vector.py

```diff
@@ -322,11 +329,18 @@ def _apply_lightning(
             elif method is not None:  # apply specialized gate
                 param = operation.parameters
                 method(wires, invert_param, param)
-            elif isinstance(operation, qml.ops.Controlled) and isinstance(
-                operation.base, qml.GlobalPhase
-            ):  # apply n-controlled gate
-                # LGPU do not support the controlled gates except for GlobalPhase
+            elif (
+                isinstance(operation, qml.ops.Controlled) and not self._mpi_handler.use_mpi
+            ):  # MPI backend does not have native controlled gates support
                 self._apply_lightning_controlled(operation)
+            elif (
+                self._mpi_handler.use_mpi
+                and isinstance(operation, qml.ops.Controlled)
+                and isinstance(operation.base, qml.GlobalPhase)
+            ):
+                raise DeviceError(
+                    "Lightning-GPU-MPI does not support Controlled GlobalPhase gates."
+                )
```

Comment:
I think it is better to move these lines inside `_apply_lightning_controlled` to avoid evaluating this condition all the time.

Reply:
I would like to keep it to ensure that MPI backend still can work with controlled gates. For the moment, if we move this error raising to the branch above, MPI backend would not work with controlled gates. You should move this branch to the _apply_lightning_controlled once there is feature parity between MPI and the serial backends